### PR TITLE
[minor_change] Add support for replacing all existing contracts with …

### LINKED ISSUE #406: Contract relationships bulk module
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Deploying a lot of consumer/provider contract relationships using current cisco.mso.mso_schema_template_anp_epg_contract module can be very slow and cause timeouts. 
It would be really nice to have contract bulk module, similar to static one mso_schema_site_anp_epg_bulk_staticport 
### New or Affected Module(s):

mso_schema_template_anp_epg_**bulk**_contract module

### MSO version and MSO Platform

Starting from NDO 3.7

### PR DESCRIPTION
**Pull request by akinross**
…new provided contracts in a single operation with one request and adding/removing multiple contracts in multiple operations with a single request in mso_schema_template_anp_epg_contract module

fixes #406

### CODE REVIEW

File: plugins/modules/mso_schema_template_anp_epg_contract.py [hunk 1]
```diff
@@ -36,31 +38,66 @@
...
+      type:
+        description:
+        - The type of the Contract.
+        type: str
+        required: true
+        choices: [ consumer, provider ]
+  contracts:
+    description:
+    - A list of Contracts associated to this EPG.
+    - This option can not be used in combination with the I(contract) option.
```
Comment (lhercot) [inline plugins/modules/mso_schema_template_anp_epg_contract.py:79]: Add some text here to explain how force_replace works with it.

**Review by lhercot — CHANGES_REQUESTED** [review]

File: plugins/modules/mso_schema_template_anp_epg_contract.py [hunk 2]
```diff
@@ -36,31 +38,68 @@
...
     description:
-    - The name of the EPG to manage.
+    - The name of the EPG.
     type: str
     required: true
+  force_replace:
+    description:
+    - Replaces all the configured contract(s) with the provided contract(s).
+    - This option can only be used in combination with the O(contracts) option.
+    - In combination with the O(state=absent) and without any contract configuration all confirgured static port(s) will be removed.
```
Comment (lhercot) [inline plugins/modules/mso_schema_template_anp_epg_contract.py:48]:
```suggestion
    - In combination with the O(state=absent) and without any contract configuration all configured static port(s) will be removed.
```

File: plugins/modules/mso_schema_template_anp_epg_contract.py [hunk 3]
```diff
@@ -36,31 +38,68 @@
...
+        description:
+        - The type of the Contract.
+        type: str
+        required: true
+        choices: [ consumer, provider ]
+  contracts:
+    description:
+    - A list of Contracts associated to this EPG.
+    - This option can not be used in combination with the O(contract) option.
+    - All configured contract(s) will be replaced with the provided contract(s) when used with O(force_replace=true).
```
Comment (lhercot) [inline plugins/modules/mso_schema_template_anp_epg_contract.py:81]:
```suggestion
    - All configured contract(s) will be replaced with the provided contract(s) when used with O(force_replace=true).
    - Only the provided contract(s) will be added, updated or removed when used with O(force_replace=false).
```

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by anvitha-jain — DISMISSED** [review]

File: plugins/modules/mso_schema_template_anp_epg_contract.py (hunk 3 above)
Comment (akinross) [reply plugins/modules/mso_schema_template_anp_epg_contract.py:81]: Also added another line to explain O(state=query) behaviour

**Review by samiib — APPROVED** [review]
LGTM

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**